viewer/core/src/public/migration/Cognite3DViewer.ts

```diff
@@ -155,6 +169,125 @@ export class Cognite3DViewer {
       new THREE.Vector3()
     )
   };
+  /**
+   * Convert pixel coordinates of the cursor to [-1,1]^2 coordinates.
+   * @param pixelX
+   * @param pixelY
+   */
+  private convertPixelCoordinatesToNormalized = (pixelX: number, pixelY: number) => {
+    const x = (pixelX / this._domElement.clientWidth) * 2 - 1;
+    const y = (pixelY / this._domElement.clientHeight) * -2 + 1;
+
+    return { x, y };
+  };
+
+  /**
+   * Calculates new target when raycaster doesn't have any intersections with the model.
+   * @param cursorPosition Cursor position for desired calculations.
+   * @param cursorPosition.x
+   * @param cursorPosition.y
+   */
+  private calculateMissedRaycast = (cursorPosition: { x: number; y: number }): THREE.Vector3 => {
+    const modelBB = this._models[0].getModelBoundingBox(new THREE.Box3()),
+      modelSize = modelBB.min.distanceTo(modelBB.max);
+
+    this.raycaster.setFromCamera(cursorPosition, this.camera);
+
+    const farPoint = this.raycaster.ray.direction
+      .clone()
+      .normalize()
+      .multiplyScalar(Math.max(this.camera.position.distanceTo(modelBB.getCenter(new THREE.Vector3())), modelSize))
+      .add(this.camera.position);
+
+    return farPoint;
+  };
+
+  /**
+   * Adds or removes event listeners for additional features of camera controls.
+   * @param removeListeners
+   */
+  private setupOtherControlsModes = (removeListeners?: boolean) => {
```

Comment:
The whole setupOtherControlsModes() and how this is handled is a bit confusing - what is "other controls modes"? Can this be changed to e.g. `setupControls(mode: ...)` ?